From ce15578fa48a21eff276ee8163d395e39785921c Mon Sep 17 00:00:00 2001
Message-Id: <ce15578fa48a21eff276ee8163d395e39785921c.1425109418.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 21 Feb 2015 12:12:34 +0530
Subject: [PATCH 1/4] SQUASHME! [VPG]: drm/i915/chv: Avoid i915 sleep before
 modeset

The i915 driver can go into runtime suspend before the first modeset &
DPST init call, to resolve it we extend the vlv implementation to
also cover cherryview.

Squash the patch to:
commit 617147c90d4a1267a5226f8ca81a66cc69c98504
Author: Sagar Kamble <sagar.a.kamble@intel.com>
Date:   Fri Dec 5 18:06:42 2014 +0530

    REVERTME [VPG]: drm/i915/vlv: Additional rpm get/put to avoid
    suspend before first modeset

Issue: GMINL-6361
Change-Id: I1241db8bc10e0ef6365fd8ba5ff9e386583317da
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c      |    2 +-
 drivers/gpu/drm/i915/intel_display.c |    3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index c4e6b96..81ef29b 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -1913,7 +1913,7 @@ int i915_driver_load(struct drm_device *dev, unsigned long flags)
 	 * Taking RPM reference here for device to stay awake till first
 	 * modeset happens. Will help initialize DPST properly.
 	*/
-	if (IS_VALLEYVIEW(dev) && !IS_CHERRYVIEW(dev))
+	if (IS_VALLEYVIEW(dev))
 		intel_runtime_pm_get(dev_priv);
 
 	i915_perfmon_setup(dev_priv);
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index e933cd4..13285f2 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12212,8 +12212,7 @@ static int __intel_set_mode(struct drm_crtc *crtc,
 	if (IS_VALLEYVIEW(dev))
 		if (dev_priv->is_first_modeset) {
 			/* This will drop reference taken in i915_driver_load */
-			if (!IS_CHERRYVIEW(dev))
-				intel_runtime_pm_put(dev_priv);
+			intel_runtime_pm_put(dev_priv);
 			dev_priv->is_first_modeset = false;
 		}
 
-- 
1.7.9.5

